### ISSUE TITLE
Cannot run - Synthetic Data

### ISSUE THREAD

**Issue by Steven-1124** [clabrugere/multitask-learning#1]
Hi there!

I am trying to replicate the work in Chapter 3.2 in the original paper. the Synthetic Data is x, y1 and y2;

The shared usage code seems to have endless errors when I input the data and cannot fix it. I suppose it may be because the input is too simple. Would you like to share the code to run on the Synthetic Data?

Obvious problem exist where:

`
model = MultiGateMixtureOfExperts(
   dim_input=num_features,
   num_tasks=num_tasks,
   num_emb=num_embeddings,
   dim_emb=32,
)
`

The input missing - dim_continuous; Also, how to decide the value of the input parameters?

Thanks so much!

**Comment by clabrugere** [comment]
Hello, so all implementations in the repo expects the input to be a tuple of `(categorical inputs, continuous inputs)` of tensors of dimensions `(batch size, num dim categorical/continuous)`. The argument `dim_continuous` is actually not used so I removed it. Keras lazily initialize weights from the first batch of data in the forward pass, so there is no need to explicitly pass it to the constructor.

I added tests you can refer to [here](https://github.com/clabrugere/multitask-learning/blob/main/tests/test_models.py#L43) and [here](https://github.com/clabrugere/multitask-learning/blob/main/tests/conftest.py#L26) is an example of how the dataset should be to work with those implementations.

As for the input parameters, I assume you mean the hyper-parameters such as number of hidden units, etc... I would suggest starting small and increase the depth and hidden dimensions if the model overfits quickly. It's very data dependent so there is no one answer that will work for every dataset. Another way would be to find papers with similar data as yours and try with the hyper-parameters they are using.

Cheers
